### PR TITLE
fix: [el-form] button type is reset and reset form

### LINKED ISSUE #3432: [Bug Report] when the native type of El button is reset, the behavior is equivalent to the form.resetfields method
### Existing Component
Yes

### Component Name
el-form el-button

### Description
手动调用form.resetFields方法可以重置v-model的值，而通过设置button的native-type="reset"后点击该按钮，虽然UI上值重置了，但是数据中并没有被重置

### PR DESCRIPTION
**Pull request by xiaoxian521**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/button/src/button.vue [hunk 1]
```diff
@@ -42,7 +43,14 @@ export default defineComponent({
       size: computed(() => elBtnGroup?.size),
     })
 
-    const handleClick = (evt: MouseEvent) => emit('click', evt)
+    const elForm = inject(elFormKey, {} as ElFormContext)
+
+    const handleClick = (evt: MouseEvent) => {
+      if (_.nativeType === 'reset') {
```
Comment (sxzz) [inline packages/components/button/src/button.vue:49]: Rename `_` to `props` please

**Comment by JeremyWuuuuu** [comment]
> Local testing is ok
> image

This could be happening from time to time but its rather rare.

**Review by sxzz — APPROVED** [review]

File: packages/components/button/src/button.vue [hunk 2]
```diff
@@ -24,25 +24,34 @@
...
+import type { ElFormContext } from '@element-plus/tokens'
 export default defineComponent({
   name: 'ElButton',
 
   props: buttonProps,
   emits: buttonEmits,
 
-  setup(_, { emit }) {
+  setup(props, { emit }) {
     const elBtnGroup = inject(elButtonGroupKey, undefined)
```
Comment (sxzz) [inline packages/components/button/src/button.vue:40]: How about setting default to `undefined`

Comment (xiaoxian521) [reply packages/components/button/src/button.vue:40]: Only write el-button type="reset", no form

File: packages/components/button/src/button.vue [hunk 3]
```diff
@@ -24,25 +24,34 @@
...
     const { size: buttonSize, disabled: buttonDisabled } = useFormItem({
       size: computed(() => elBtnGroup?.size),
     })
 
-    const handleClick = (evt: MouseEvent) => emit('click', evt)
+    const elForm = inject(elFormKey, {} as ElFormContext)
```
Comment (JeremyWuuuuu) [inline packages/components/button/src/button.vue:47]:
```suggestion
const elForm = inject(elFormKey, undefined)
```

Comment (xiaoxian521) [reply packages/components/button/src/button.vue]: down

**Review by jw-foss — APPROVED** [review]
LGTM

File: packages/components/button/src/button.vue (hunk 3 above)
Comment (sxzz) [reply packages/components/button/src/button.vue]: BTW we should remove `isEmpty`...

Comment (xiaoxian521) [reply packages/components/button/src/button.vue]: ok i will pay attention

**Pull request merged**